src/langapi/mode.cpp

```diff
@@ -103,30 +134,25 @@ std::unique_ptr<languaget> get_language_from_filename(
 {
   std::size_t ext_pos=filename.rfind('.');
 
-  if(ext_pos==std::string::npos)
+  if(ext_pos == std::string::npos)
     return nullptr;
 
   std::string extension=
     std::string(filename, ext_pos+1, std::string::npos);
 
-  if(extension=="")
+  if(extension.empty())
     return nullptr;
 
-  for(languagest::const_iterator
-      l_it=languages.begin();
-      l_it!=languages.end();
-      l_it++)
+  for(const auto &language_info : languages)
   {
+    const auto &extensions = language_info->extensions();
     #ifdef _WIN32
-    for(std::set<std::string>::const_iterator
-        e_it=l_it->extensions.begin();
-        e_it!=l_it->extensions.end();
-        e_it++)
-      if(_stricmp(extension.c_str(), e_it->c_str())==0)
-        return l_it->factory();
+    for(const auto &ext : extensions)
```

Comment:
to avoid having two variables called `extension` (one abbreviated) could you rename either the original or the new one to something more descriptive? (e.g. `language_extension` vs `file_extension` perhaps?)